### ISSUE TITLE
[useScrollLock] Glitch in Safari when opening while pinch-zoomed

### ISSUE THREAD

**Issue by atomiks** [mui/base-ui#1130]
Started happening after 83b738d26f9f909def971e5542cb145fc25a6a22

https://github.com/user-attachments/assets/06b5f745-2143-4b1f-b59b-8106a68e3d6b

**Comment by atomiks** [comment]
This is likely caused by `useScrollLock` since it happens with Dialog too. I think Safari needs to read `visualViewport` values in order for `position: fixed` calculations to be correct.

**Comment by michaldudak** [comment]
It's much better now, but there's still a slight movement to the bottom-right when the menu popup opens while zoomed in. IMO we can live with it, but let's see if it's easily solvable.

**Comment by atomiks** [comment]
This issue is no longer relevant after #1159
